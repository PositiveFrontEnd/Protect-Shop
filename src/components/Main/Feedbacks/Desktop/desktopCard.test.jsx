import React from "react";
import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import DesktopCard from "./desktopCard";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

describe("Тестування компонента  desktopCard", () => {
    test("Знімок", () => {
        const mockStore = configureStore();
        const initialState = {
            home: {
                isAnimation: false,
                inputText: "",
                filterHomePage: [],
                marker: null,
            },
            user: {
                token: "token",
                registrationModal: false,
                authorizationStatus: null,
                registrationStatus: "some text",
                registrationData: {},
                correctData: "some text",
                isAdmin: false,
            },
            products: {
                products: [],
                productsByCategory: [],
                productsByType: [],
                oneProduct: {},
                ThreeProducts: [],
                ThreeColors: [],
                filters: [],
                twelveProducts: [],
                filtersParameters: {},
                twelveFiltersProducts: [],
                searchProducts: [],
                youSee: [],
                previewProductInfo: {},
            },
            order: {
                orderState: [],
                productsForOrderGuest: {},
                infoForOrderGuest: [],
                deliveryForOrderGuest: {},
                order: {
                    letterSubject: "Thank you for order!",
                },
                promoCodePrice: "",
                orderGuest: { letterSubject: "Thank you for order!" },
            },
            basket: {
                basket: [],
                priceGuest: "''",
                guestBasket: {},
            },
            catalogue: {
                sortCatalogue: "some text",
                loadFilter: false,
            },
            favorite: {
                favorite: [],
                favoriteForCustomer: [],
                guestFavorite: [],
            },
            message: {
                letters: [],
                letterAll: [],
                importantLetters: []
            },
            shopComments: {
                createShopComment: [],
                getShopComments: [],
            },
        };
        const data = {
            avatar: "https://anybag.ua/upload/resize_cache/iblock/c36/450_450_140cd750bba9870f18aada2478b24840a/f439vexpgnfug8w4syc2marwqqde4u3q.jpeg",
            name: "some",
            comment: "some",
            likes: 1,
            background: "string"
        }

        const store = mockStore(initialState);
        const desktopCard = render(
            <Provider store={store}>
                <BrowserRouter>
                    <DesktopCard data={data} />
                </BrowserRouter>
            </Provider>
        );

        expect(desktopCard).toMatchSnapshot();
    });
});
