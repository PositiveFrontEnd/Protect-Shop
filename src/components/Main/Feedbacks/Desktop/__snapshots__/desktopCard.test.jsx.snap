// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  desktopCard > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="feedbacks__card"
      >
        <div
          class="no__avatar__wrapper"
        >
          <p
            class="avatar__letter"
          />
        </div>
        <div
          class="feedbacks__stars"
        >
          <svg
            class=" star inactive"
            fill="none"
            height="20"
            viewBox="0 0 12 12"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_1704_1143)"
            >
              <path
                d="M6.20634 0.5L7.55342 4.6459H11.9127L8.38597 7.2082L9.73305 11.3541L6.20634 8.7918L2.67963 11.3541L4.02671 7.2082L0.5 4.6459H4.85925L6.20634 0.5Z"
                fill="#FDE82A"
              />
            </g>
            <defs>
              <clippath
                id="clip0_1704_1143"
              >
                <rect
                  fill="white"
                  height="12"
                  width="12"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <p
          class="feedbacks__card__name"
        />
        <p
          class="feedbacks__card__description"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="feedbacks__card"
    >
      <div
        class="no__avatar__wrapper"
      >
        <p
          class="avatar__letter"
        />
      </div>
      <div
        class="feedbacks__stars"
      >
        <svg
          class=" star inactive"
          fill="none"
          height="20"
          viewBox="0 0 12 12"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_1704_1143)"
          >
            <path
              d="M6.20634 0.5L7.55342 4.6459H11.9127L8.38597 7.2082L9.73305 11.3541L6.20634 8.7918L2.67963 11.3541L4.02671 7.2082L0.5 4.6459H4.85925L6.20634 0.5Z"
              fill="#FDE82A"
            />
          </g>
          <defs>
            <clippath
              id="clip0_1704_1143"
            >
              <rect
                fill="white"
                height="12"
                width="12"
              />
            </clippath>
          </defs>
        </svg>
      </div>
      <p
        class="feedbacks__card__name"
      />
      <p
        class="feedbacks__card__description"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
